desc/compute/_neoclassical.py

```diff
@@ -0,0 +1,275 @@
+"""Compute functions for neoclassical transport.
+
+Notes
+-----
+Some quantities require additional work to compute at the magnetic axis.
+A Python lambda function is used to lazily compute the magnetic axis limits
+of these quantities. These lambda functions are evaluated only when the
+computational grid has a node on the magnetic axis to avoid potentially
+expensive computations.
+"""
+
+from functools import partial
+
+from orthax.legendre import leggauss
+from quadax import simpson
+
+from desc.backend import imap, jit, jnp
+
+from .bounce_integral import bounce_integral, get_pitch
+from .data_index import register_compute_fun
+from .utils import safediv
+
+
+def _vec_quadax(quad, **kwargs):
+    """Vectorize an adaptive quadrature method from quadax.
+
+    Parameters
+    ----------
+    quad : callable
+        Adaptive quadrature method matching API from quadax.
+
+    Returns
+    -------
+    vec_quad : callable
+        Vectorized adaptive quadrature method.
+
+    """
+
+    def vec_quad(fun, interval, B_sup_z, B, B_z_ra, arg1, arg2):
+        return quad(fun, interval, args=(B_sup_z, B, B_z_ra, arg1, arg2), **kwargs)[0]
+
+    vec_quad = jnp.vectorize(
+        vec_quad, signature="(2),(m),(m),(m),(m),(m)->()", excluded={0}
+    )
+    return vec_quad
+
+
+def _get_pitch(grid, min_B, max_B, num, for_adaptive=False):
+    """Get points for quadrature over velocity coordinate.
+
+    Parameters
+    ----------
+    grid : Grid
+        The grid on which data is computed.
+    min_B : jnp.ndarray
```

Comment:
here min_B and max_B are ndarray of size `grid.num_nodes`?

Reply:
and are they max/min B on a flux surface?

Reply:
yes its `data["max_tz |B|"]` and `data["min_tz |B|"]`